src/azul/service/responseobjects/manifest_service.py

```diff
@@ -0,0 +1,60 @@
+import base64
+import json
+
+from azul import config
+from azul.service.responseobjects.step_function_helper import StateMachineError, StepFunctionHelper
+
+
+class ManifestService:
+    """
+    Class containing logic for starting and checking the status of manifest generation jobs
+    """
+    manifest_endpoint = '/repository/manifest/files'
+
+    step_function_helper = StepFunctionHelper()
+
+    def encode_params(self, params, encoding='utf-8'):
+        return base64.urlsafe_b64encode(bytes(json.dumps(params), encoding=encoding)).decode(encoding)
+
+    def decode_params(self, token, encoding='utf-8'):
+        return json.loads(base64.urlsafe_b64decode(token).decode(encoding))
+
+    def start_manifest_generation(self, filters):
+        """
+        Start the execution of a state machine generating the manifest
+
+        :param filters: filters to use for the manifest
+        :return: The id of the execution of the state machine
+        """
+        return self.step_function_helper.start_execution(
+            config.manifest_state_machine_name,
+            execution_input={'filters': filters}
+        )['executionArn'].split(':')[-1]
+
+    def get_manifest_status(self, params, wait, local=False):
+        """
+        Get the status of a manifest generation job (in progress, success, or failed)
+
+        :param params: parameters to use when checking the status of the execution
+        :param wait: wait step to use as a param in the retry url
+        :param local: True if running chalice locally, False otherwise
+        :return: dict containing status of the job and a redirect url for the client
+        """
+        execution = self.step_function_helper.describe_execution(
+            config.manifest_state_machine_name, params['execution_id'])
+
+        if execution['status'] == 'SUCCEEDED':
+            execution_output = json.loads(execution['output'])
+            return {
+                'Status': 302,
+                'Location': execution_output['Location']
+            }
+        elif execution['status'] == 'RUNNING':
+            wait_times = [1, 1, 2, 6, 10]
+            base_url = 'http://localhost:8000' if local else config.service_endpoint()
```

Comment:
See if you can avoid hard-coding the endpoint using `app.current_request.headers['host']`.

This means you'd probably have to pass `app.current_request` from `app.py` down to the service but I think that's OK for this particular purpose. I still think `app.py` should extract the URL parameters and pass them explicitly as separate method arguments.

Reply:
That works.  Just need to check the `X-Forwarded-Proto` header for http vs. https and assume if the header is missing then it's http.

Reply:
I meant only when `local` is `True`. When `local` is `False`, the existing method to synthesize the URL is sufficient.

Reply:
Wouldn't it be simpler to remove local and always use `app.current_request.headers['host']` to create the URL.  That way the the URL can be created in `app.py` and passed here as an argument and there's just one way to create the URL.

Reply:
Is there some case where `app.current_request.headers['host']` doesn't give us what we want?

Reply:
One thing that just occured to me is that if we don't use a `local` param we can't really test the case where the path is the actual deployed path.

Reply:
No, you're right. I was worried about someone forging the `Host` header but I checked that API Gateway validates the header so it should be safe to use.